### ISSUE TITLE
When round restarts, UI is permanently empty B: 50

### ISSUE THREAD

**Issue by chairbender** [unitystation/unitystation#1664]
# Description
After the round restarts and you select a job and spawn, the UI is shown as empty and never appears to update even when picking things up and even though your sprite is shown as holding something / having clothes.

It also seems that is becomes impossible to shoot, throw, or melee. Basically becomes impossible to use any item in your hands.

Expected behavior is that the UI shows up just like it does when you first join a game and that you can use items and the UI is updated when you pickup / drop stuff.

(Currently only discovered on headless)

**Comment by DooblyNoobly** [comment]
Bounty of 50 usd will be awarded to the contributor on a successful merge of a solution to this bug

**Comment by DooblyNoobly** [comment]
looks like #1677 has fixed this, I just did a test on steam server (Spawned as nuke ops, blew up station and when round restarted everything was fine with the UI slots)

**Comment by fomalsd** [comment]
are you sure it wasn't just because of messed up server back then?

**Comment by DooblyNoobly** [comment]
well Inventory manager errors probably contributed to it on the server. As was pointed out in the note fixes `lambda expressions and null propagation operators dont get along` which seems to hold true when looking for explanations why, I found this on stack overflow: `tree lambdas (unlike delegate lambdas) are interpreted by already existing LINQ providers which don't yet support null propagating.` which makes sense. Learn something new every day.

Also I just done another test to make sure and its working, this time I called restart-round from admin portal
